droidlet/dashboard/web/src/components/Memory2D/MemoryMapTable.js

```diff
@@ -52,71 +119,139 @@ const useStyles = makeStyles({
  *                            onTableDone: event handler for after user is finished with table.
  */
 export default function MemoryMapTable(props) {
-  const classes = useStyles();
+  /*
+  The editManager handles the state of the table in the form of a dictionary of the rows.
+  Keys are row attributes.
+  Value is object [
+    value: <value>,
+    valueType: <value_type>,
+    status: "same" or "changed" or "error" [or "new" or "deleted"] [for future],
```

Comment:
If we want to include functionality for adding/removing tags, we can alter "status" and handle with ManualEditsPerception module in backend